### PR TITLE
[WIP] feat(backend): API for Merge Changes from Master

### PR DESCRIPTION
**Pull request by dewen**
## Changes

- Create API endpoint to return master to upstream branch merge conflict info.
- Add tests

## Test Instructions

- Clone and checkout the dev branch.
- npm run setup
- cd examples/test-site
- npm run dev-backend
- Make some changes and push to remote upstream branch.
- Switch to master branch and make different changes to the same files as on dev branch. This will create conflict when merging from master to upstream dev branch.
- Run the web API with a http client, i.e. POSTMAN or curl, i.e.
```
curl --request GET --url http://localhost:8006/___backend/changes/conflicts
```

Expects: conflict files reported from response.

## Related Issues

Fixes: #313

### CODE REVIEW

File: packages/bodiless-backend/src/GitCmd.js [hunk 1]
```diff
@@ -24,13 +24,19 @@ class GitCmd {
...
     this.files = [];
+    this.options = {};
   }
 
   add(...params) {
     this.params.push(...params);
     return this;
   }
 
+  set(option) {
```
Comment (wodenx) [inline packages/bodiless-backend/src/GitCmd.js:35]: should be options (plural)

File: packages/bodiless-backend/src/git.js [hunk 2]
```diff
@@ -85,21 +99,104 @@ const getChanges = async () => {
...
+
+/**
+ * Get origin master to upstream branch merge conflict status.
+ *
+ * @return {object} Results.
+ */
+const getConflicts = async () => {
+  // const remoteUrl = await getRemote('origin');
+  const logger = new Logger('BACKEND');
+  const directory = path.resolve(process.env.BODILESS_BACKEND_TMP || '/tmp', v1());
```
Comment (wodenx) [inline packages/bodiless-backend/src/git.js:134]: consider using [`os.tmpdir()`](https://nodejs.org/api/os.html#os_os_tmpdir) here.

File: packages/bodiless-backend/package.json [hunk 3]
```diff
@@ -12,7 +12,9 @@
     "dotenv": "^8.2.0",
     "express": "^4.17.1",
     "formidable": "^1.2.1",
-    "tmp": "^0.0.33"
+    "rimraf": "^2.6.3",
```
Comment (wodenx) [inline packages/bodiless-backend/package.json:15]: No changes required in root package-lock.json due to adding these dependencies?

File: packages/bodiless-backend/README.md [hunk 4]
```diff
@@ -13,3 +13,36 @@ You can start backend server by running `node ./path/to/server.js`. Backend serv
 * `BODILESS_BACKEND_EXTENDED_LOGGING_ENABLED` - Whether or not extended logging is enabled. Defaults to `0`. Please note that this feature requires `morgan` and `morgan-body` packages. If you plan to use this feature outside of monorepo, please ensure those dependencies are installed.
 
 Please note that `BODILESS_BACKEND_DATA_FILE_PATH`, `BODILESS_BACKEND_STATIC_PATH` and `BODILESS_BACKEND_COMMIT_ENABLED` are defined in `@bodiless/gatsby-theme-bodiless` by default.
+
+## Web API
+
+Backend package provides a set of web API endpoints to allow developers working on git workflow.
```
Comment (wodenx) [inline packages/bodiless-backend/README.md:19]: Please add a statement here that this is a private API and that not all routes are currently documented.

File: packages/bodiless-backend/package.json (hunk 3 above)
Comment (dewen) [reply packages/bodiless-backend/package.json:15]: These packages should be in root package-lock.json already. Tested with new setup.

File: packages/bodiless-backend/src/git.js [hunk 5]
```diff
@@ -27,6 +28,7 @@ const getCurrentBranch = async () => {
 
 /**
  * Verify the existence of an upstream branch.
+ * @todo: replace with getUpstreamTrackingBranch?
```
Comment (wodenx) [inline packages/bodiless-backend/src/git.js:31]: I think let's leave as is for now and reconsider in scope of the p.sh bug.

**Pull request merged**